### ISSUE TITLE
error GraphQL Error Field [type Image] must not have a selection since type "String" has no subfields. 

### ISSUE THREAD

**Issue by ChandlerMBing** [netlify/netlify-cms#2050]
**Describe the bug**

After adding or changing the picture in a `Image` widget inside the cms, the local environment fails to build while production builds just fine
**To Reproduce**

1. /admin
2. Page with Image widget
3. Upload or change picture
4. Pull changes from repo
5. See error in local environment
6. Production builds just fine

**Expected behaviour**
local environment successfully builds the site
**Screenshots**

Error in browser
<img width="1052" alt="error" src="https://user-images.githubusercontent.com/23351348/52126297-4259a380-262f-11e9-82e0-40d7b0db8231.png">

Two queries in graphiql one is treating img as string the other is copy of the query from the template file
<img width="1093" alt="graphiql-query" src="https://user-images.githubusercontent.com/23351348/52126014-754f6780-262e-11e9-8d30-761756374577.png">
<img width="1083" alt="graphiql-query-error" src="https://user-images.githubusercontent.com/23351348/52126016-754f6780-262e-11e9-9c0a-6f9ebb7d6178.png">

Here's the frontmatter for index.md
<img width="588" alt="frontmatter" src="https://user-images.githubusercontent.com/23351348/52125848-083bd200-262e-11e9-9b9e-9815fcea1807.png">

My File Structure [pretty much standard gatsby]
<img width="492" alt="file-structure" src="https://user-images.githubusercontent.com/23351348/52125872-1853b180-262e-11e9-8d48-ed56642dce15.png">

Gatsby-source-filesystem settings for images and uploads
<img width="535" alt="file-system-images" src="https://user-images.githubusercontent.com/23351348/52125950-4802b980-262e-11e9-84bb-2979c6c7d426.png">
<img width="540" alt="file-system-uploads" src="https://user-images.githubusercontent.com/23351348/52125952-489b5000-262e-11e9-9d1a-e349fcf9f016.png">

Query in template file
<img width="666" alt="template-query" src="https://user-images.githubusercontent.com/23351348/52125985-5c46b680-262e-11e9-965f-05c8d0c2d703.png">

**Applicable Versions:**

 - Netlify CMS version: 2.3.2
 - Git provider: GitLab
 - OS: OSX Mojave
 - Browser version: Up-to-date chrome
 - Node.JS version: 10.13.0
 - npm: 6.4.1

**CMS configuration**

```yaml
backend:
  name: git-gateway
  branch: master

media_folder: static/img
public_folder: /img
logo_url: https://proxy.duckduckgo.com/iu/?u=http%3A%2F%2Ftous-logos.com%2Fwp-content%2Fuploads%2F2017%2F09%2FLogo-IKEA.jpg&f=1
display_url: http://localhost:8000/
collections:
  - name: 'pages'
    editor:
      preview: false
    label: 'Pages'
    create: true
    files:
      # index start
      - file: 'src/pages/index.md'
        label: 'Home'
        name: 'home'
        fields:
          - {
            label: 'Template Key',
            name: 'templateKey',
            widget: 'hidden',
            default: 'index-page',
          }
          - { label: Title, name: title, widget: string}
          - { label: Hero, name: hero__image, widget: image}
          - { label: Body, name: body, widget: markdown}
      # index end
```

**Additional context**
Sometimes it works sometimes it doesn't, I tried re-running gatsby develop, changing gatsby filesystem plugin path for images from src to static, deleting cache and everything else in the `application` tab in dev tools. Tried to find something online but couldn't so it's either some weird obscure bug that no-one has ever encountered or, hopefully, I messed something up somewhere. 🙄

**Comment by Undistraction** [comment]
@ChandlerMBing This is a Gatsby issue, not related to Netlify CMS

**Comment by ChandlerMBing** [comment]
Thanks for explaining the issue, I shall brace myself for the rabbithole then.

**Comment by Undistraction** [comment]
@ChandlerMBing My advice is to concentrate on getting a single image working. That error is occurring because the image file doesn't exist at the path, so it is left as a string instead of being transformed into an image node. While trying to get it working, make sure you keep clearing Gatsby's cache so that it re-evaluates any changes. I would advise you first make sure images are loading and displaying properly in the CMS, then move to getting them working in Gatsby. Much easier than doing this the other way around because there is much less you can tweak in the CMS. Also be aware you will need to use a separate function from the library I linked to if you want to use images within your markdown (not just as field values). Good luck. It does work, but you'll probably pull some hair out on the journey.

**Comment by ChandlerMBing** [comment]
@Undistraction  That's the thing, in production and in the cms it works just fine, it only breaks when I run gatsby develop.

**Comment by erquhart** [comment]
@ChandlerMBing did you figure this out?

**Comment by ChandlerMBing** [comment]
@erquhart Yup, it was 100% user error, I didn't clean the cache because I was running the localhost with `gatsby develop` rather than `npm start`. I haven't had this problem since. Sorry, I should've said that before closing, I forgot. 😇

**Comment by dbertella** [comment]
Really? I have the same error but in prod, the build failed every time I pick an image from the media library, it works fine if I upload a new image, because in the second case the image is saved with a relative path as gatsby probably expected.

**Comment by amykapernick** [comment]
I've been having the same issues on Gatsby with headless WP and Gatsby with Netlify CMS. Was working fine before but not now

**Comment by Zellement** [comment]
I updated to 2.8.0 yesterday to fix another issue I was told about on Gitter, and this has since broken my images too.

**Comment by erquhart** [comment]
Should be fixed in `netlify-cms@2.9.0`.
